src/odemis/util/spot.py

```diff
@@ -250,43 +252,44 @@ def MaximaFind(image, qty, len_object=18):
     filtered = BandPassFilter(image, 1, len_object)
 
     # dilation
-    s = _CreateSEDisk(len_object // 2)
-    dilated = ndimage.grey_dilation(filtered, footprint=s)
+    structure = _CreateSEDisk(len_object // 2)
+    dilated = cv2.dilate(filtered, structure, iterations=1)
 
     # find local maxima
     binary = (filtered == dilated)
 
     # thresholding
     qradius = max(len_object // 4, 1)
-    BWdil = ndimage.binary_dilation(binary, structure=_CreateSEDisk(qradius))
+    structure = _CreateSEDisk(qradius)
+    BWdil = cv2.dilate(numpy.array(binary).astype(numpy.uint8), structure)
 
     labels, num_features = ndimage.label(BWdil)
     index = numpy.arange(1, num_features + 1)
     mean = ndimage.mean(filtered, labels=labels, index=index)
     si = numpy.argsort(mean)[::-1][:qty]
     # estimate spot center position
-    pos = numpy.array(ndimage.center_of_mass(filtered, labels=labels,
-                                          index=index[si]))[:, ::-1]
+    pos = numpy.array(ndimage.center_of_mass(filtered, labels=labels, index=index[si]))[:, ::-1]
+
     if numpy.any(numpy.isnan(pos)):
         pos = pos[numpy.any(~numpy.isnan(pos), axis=1)]
         logging.debug("Only %d maxima found, while expected %d", len(pos), qty)
     # improve center estimate using radial symmetry method
     w = len_object // 2
-    refined_position = numpy.zeros_like(pos)
+    refined_center = numpy.zeros_like(pos)
+    pos = numpy.rint(pos).astype(numpy.int16)
     for idx, xy in enumerate(pos):
-        _ij = numpy.rint(xy).astype(int)
-        i0, j0 = _ij - w + 1
-        i1, j1 = _ij + w
-        # if spot is close to edge of image i0 and j0 can be smaller than 0, crop so the spot is still in the center.
-        if i0 < 0:
-            i1, j1 = numpy.array([i1, j1]) + i0
-            i0, j0 = numpy.array([i0, j0]) - i0
-        if j0 < 0:
-            i1, j1 = numpy.array([i1, j1]) + j0
-            i0, j0 = numpy.array([i0, j0]) - j0
-        spot = filtered[j0:j1, i0:i1]
-        refined_position[idx] = numpy.rint(xy) + numpy.array(FindCenterCoordinates(spot))
-
+        x_start, y_start = xy - w + 1
+        x_end, y_end = xy + w
+        # if spot is close to edge of image x and y can be smaller than 0, crop so the spot is still in the center.
+        if x_start < 0:
+            x_end = x_end + x_start
+            x_start = 0
+        if y_start < 0:
+            y_end = y_end + y_start
+            y_start = 0
+        spot = filtered[y_start:y_end, x_start:x_end]
```

Comment:
@pieleric updated the pull request with your comments. I'm still a bit confused about the indexing with y, x here instead of x, y but I think it's correct.